mkdocs/overrides/devbook/transactions/bonded-aggregate.py

```
@@ -0,0 +1,257 @@
+	hash_lock = facade.transaction_factory.create({
+		'type': 'hash_lock_transaction_v1',
+		'signer_public_key': account_a_key_pair.public_key,
+		'deadline': timestamp.add_hours(2).timestamp,
+		'mosaic': {
+			'mosaic_id': generate_mosaic_alias_id('symbol.xym'),
+			'amount': 10_000_000  # 10 XYM deposit
+		},
+		'duration': 100,  # Lock duration in blocks
+		'hash': bonded_hash
+	})
+	hash_lock.fee = Amount(fee_mult * hash_lock.size)
+
+	# Sign hash lock
+	print('[Account A] Signing the hash lock...')
+	hash_lock_signature = facade.sign_transaction(
+		account_a_key_pair, hash_lock)
+	hash_lock_payload = facade.transaction_factory.attach_signature(
+		hash_lock, hash_lock_signature)
+	hash_lock_hash = facade.hash_transaction(hash_lock)
+	print(f'Hash lock transaction hash: {hash_lock_hash}')
+
+	# Announce hash lock and wait for confirmation
+	announce_transaction(hash_lock_payload, '/transactions', 'Hash lock')
+	wait_for_status(hash_lock_hash, 'confirmed', 'Hash lock')
+
+	# Announce bonded aggregate and wait for partial status
+	announce_transaction(
+		bonded_json_payload, '/transactions/partial',
+		'Bonded aggregate transaction'
+	)
+	wait_for_status(
+		bonded_hash, 'partial',
+		'Bonded aggregate transaction'
+	)
+
+	# --- ACCOUNT B (Cosigner) ---
+	# Retrieves partial transactions waiting for signature
+	partial_path = f'/transactions/partial?address={account_b_address}'
+	print(
+		'[Account B] Checking for partial transactions from '
+		'/transactions/partial'
+	)
+	with urllib.request.urlopen(f'{NODE_URL}{partial_path}') as response:
+		partial_txs = json.loads(response.read().decode())
+		if not partial_txs['data']:
+			raise Exception('No partial transactions found')
+
+	print(f'Found {len(partial_txs["data"])} partial transaction(s)')
+
+	# Find the transaction matching our expected hash
+	partial_tx_hash = None
+	for tx in partial_txs['data']:
+		if tx['meta']['hash'] == str(bonded_hash):
+			partial_tx_hash = tx['meta']['hash']
```

Wait. So `bonded_hash` and `partial_tx_hash` are the same thing?
This is very confusing, I assumed they were different hashes and was reading the spec trying to find what they are :D

You can raise an exception if `bonded_hash` is not found in `partial_txs`, and proceed otherwise.
But don't use another var holding the same hash.